agent.go

```diff
@@ -133,6 +133,17 @@ type agent interface {
 	// to handle all other Agent interface methods.
 	init(pod *Pod, config interface{}) error
 
+	// vmURL returns the agent URL exposed by the hypervisor. This URL is
+	// directly connected to the agent running inside the virtual machine.
```

Comment:
Could we please add some more information here: In particular, could we clearly explain that the VM URL needs to be built and passed to the hypervisor by the agent.

Reply:
Yes, I'll try to be more explicit and verbose about that.